test/test-cmd.sh

```diff
@@ -0,0 +1,38 @@
+#! /bin/bash
+
+set -e
+
+# start server
+./server &
+# sleep a bit to let server spin up
+sleep 2
```

Comment:
I tend to think that busy pulling with a timeout is a bit over-skill just for this little test script.

Reply:
Yes, it's a very simple script at present, so I don't think this is a "blocker" to get this landed. It's just something we need to think about going forward as the script grows to avoid odd failures on (very ;) slow systems.

Reply:
Overall the `proxy` program will be very simple & small. But yeah, who knows ;)

Reply:
Yeah, we have seen problems with fixed timeouts (even quite large timeouts) when running CI tests on cloud based VM machines - so we've become wary of fixed timeouts and try to avoid them when we can. Flaky CI failures can become the bane of our lives otherwise :-)